Mutation improver

andyMrtnzP opened the pull request:
## Description

Modified MutationObserver in `content.js` to filter DOM mutations before iterating over them.

## Browsers

Tested and working as expected on:

- Google Chrome (96.0.4664.45).
- Mozilla Firefox (94.0.2).

wanhose commented on scripts/content.js:
```diff
@@ -139,13 +139,13 @@ const observer = new MutationObserver((mutations, instance) => {
-        if (valid && node.matches(selectors)) node.outerHTML = "";
-      });
-    });
+    const nodes = mutations.map((mutation) => [...mutation.addedNodes])
```
I really like the "more functional" approach but it seems a less performant way to iterate these arrays. Previously we had only two iterations using two `for each` loops and now we have four: `map`, `flat`, `filter` and finally a `for of` loop.

andyMrtnzP replied:
Oh, I clearly see what you mean. So sorry. I'll revert those changes

wanhose commented on scripts/content.js:
```diff
@@ -139,13 +139,13 @@ const observer = new MutationObserver((mutations, instance) => {
+                           .flat()
+                           .filter((node) => check(node));
+    
+    for (const node of nodes) {
```
Benchmarks say that a `for of` loop is more performant than a `for each` loop in some browsers. How about changing these two `for each` by two `for of` loops?

andyMrtnzP replied:
Yup, will do it!

wanhose reviewed (changes requested):
Welcome to the community @andyMrtnzP, 

You're the first code contributor to the project. I really like your suggestion but I have some comments. Please take a look. 😁

andyMrtnzP commented:
Reverted last commit. I also added element selectors for some websites I encountered that the extension doesn't cover yet (Libreriaberkana, Verso, Medium, Tidal, Medium, Mercadolibre and others)

wanhose commented:
Wooow! Amazing job @andyMrtnzP!

I will check these new selectors and then I will merge everything into our next release 🚀

wanhose commented:
Hi again @andyMrtnzP!

Could you resolve conflicts and squash all your commits into just one?

wanhose commented:
Hi again @andyMrtnzP!

Could you resolve conflicts, squash all your commits into just one and set the target branch to `5.4.4` branch?

wanhose commented:
Hi again @andyMrtnzP!

I'm going to close the pull request because I have already copied your commits to the release branch. You don't have to do nothing else.

Thank you again for your contribution.
